istioctl/pkg/install/pre-check.go

```diff
@@ -0,0 +1,159 @@
+)
+
+type targetResource struct {
+	namespace string
+	group     string
+	version   string
+	name      string
+}
+
+func installPreCheck(istioNamespaceFlag *string, restClientGetter resource.RESTClientGetter, writer io.Writer) error {
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "Checking the cluster to make sure it is ready for Istio installation...\n")
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "kubernetes-api\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+	kubeClient, err := createKubeClient(restClientGetter)
+	if err != nil {
+		return fmt.Errorf("failed to initialize the k8s client: %v", err)
+	}
+	fmt.Fprintf(writer, "can initialize the k8s client\n")
+	v, err := kubeClient.Discovery().ServerVersion()
+	if err != nil {
+		return fmt.Errorf("failed to query the Kubernetes API Server: %v", err)
+	}
+	fmt.Fprintf(writer, "can query the Kubernetes API Server\n")
+
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "kubernetes-version\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+
+	if !checkKubernetesVersion(v) {
+		return fmt.Errorf("The Kubernetes API version: %v is lower than the minimum version: "+minK8SVersion, v)
+	}
+	fmt.Fprintf(writer, "is running the minimum Kubernetes API version\n")
+
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "Istio-existence\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+	ns, _ := kubeClient.CoreV1().Namespaces().Get(*istioNamespaceFlag, meta_v1.GetOptions{})
+	if ns != nil {
+		return fmt.Errorf("control plane namespace: %v already exist", *istioNamespaceFlag)
+	}
+	fmt.Fprintf(writer, "control plane namespace does not already exist\n")
+
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "kubernetes-setup\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+	Resources := make([]targetResource, 0)
+	Resources = append(Resources, targetResource{"", "", "v1", "Namespace"})
+	Resources = append(Resources, targetResource{*istioNamespaceFlag, "rbac.authorization.k8s.io", "v1beta1", "ClusterRole"})
+	Resources = append(Resources, targetResource{*istioNamespaceFlag, "rbac.authorization.k8s.io", "v1beta1", "ClusterRoleBinding"})
+	Resources = append(Resources, targetResource{*istioNamespaceFlag, "apiextensions.k8s.io", "v1beta1", "CustomResourceDefinition"})
+	Resources = append(Resources, targetResource{*istioNamespaceFlag, "rbac.authorization.k8s.io", "v1beta1", "Role"})
+	Resources = append(Resources, targetResource{*istioNamespaceFlag, "", "v1", "ServiceAccount"})
+	Resources = append(Resources, targetResource{*istioNamespaceFlag, "", "v1", "Service"})
+	Resources = append(Resources, targetResource{*istioNamespaceFlag, "extensions", "v1beta1", "Deployments"})
+	Resources = append(Resources, targetResource{*istioNamespaceFlag, "", "v1", "ConfigMap"})
+	for _, targetResource := range Resources {
+		err = checkCanCreateResources(kubeClient, targetResource)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(writer, "Can create %v \n", targetResource.name)
```

Comment:
A shorter message is preferable.  9 lines of output is a lot.  "Can create necessary Kubernetes configuration" or similar summary line instead of details.

Reply:
done